src/measure_theory/interval_integral.lean

```diff
@@ -0,0 +1,405 @@
+begin
+  cases le_total a b with ab ba,
+  { rw [integral_of_le ab, integral_of_le' ab, interval_swap, neg_neg] },
+  { rw [integral_of_le ba, integral_of_le' ba, interval_swap] }
+end
+
+lemma integral_undef (h : ¬ (measurable_on a..b f ∧ integrable_on a..b f)) :
+  (∫ x in a...b, f x) = 0 :=
+begin
+  split_ifs with hab,
+  { simp only [interval_of_le hab] at *, exact integral_undef h },
+  { simp only [interval_of_not_le hab] at *, rw [integral_undef h, neg_zero] }
+end
+
+lemma integral_congr (h : ∀ x, x ∈ a..b → f x = g x) : (∫ x in a...b, f x) = ∫ x in a...b, g x :=
+begin
+  split_ifs with h',
+  { simp only [interval_of_le h'] at *,
+    exact integral_on_congr h },
+  { simp only [interval_of_not_le h'] at *,
+    rw integral_on_congr h }
+end
+
+lemma integral_congr_ae (h : ∀ₘ x, x ∈ a..b → f x = g x)
+  (hfm : measurable_on a..b f) (hgm : measurable_on a..b g) :
+  (∫ x in a...b, f x) = (∫ x in a...b, g x) :=
+begin
+  split_ifs with h,
+  { simp only [interval_of_le h] at *,
+    rw integral_on_congr_of_ae_eq,
+    assumption' },
+  { simp only [interval_of_not_le h] at *,
+    rw integral_on_congr_of_ae_eq,
+    assumption' }
+end
+
+lemma integral_add (hfm : measurable_on a..b f) (hfi : integrable_on a..b f)
+  (hgm : measurable_on a..b g) (hgi : integrable_on a..b g) :
+  (∫ x in a...b, f x + g x) = (∫ x in a...b, f x) + (∫ x in a...b, g x) :=
+begin
+  split_ifs with h,
+  { simp only [interval_of_le h] at *,
+    rw [integral_on_add], assumption' },
+  { simp only [interval_of_not_le h] at *,
+    rw [← neg_add, integral_on_add], assumption' }
+end
+
+lemma integral_const {c : β} : (∫ x in a...b, c) = (b - a) • c :=
+begin
+  split_ifs with h,
+  { rw [integral_on_const, volume_interval, ← ennreal.of_real, ennreal.to_real_of_real, abs_of_nonneg],
+    exact sub_nonneg_of_le h, exact abs_nonneg _, exact is_measurable_interval },
+  { rw [integral_on_const, volume_interval, ← ennreal.of_real, ennreal.to_real_of_real,
+      abs_of_nonpos, ← neg_smul, neg_neg],
+    exact sub_nonpos_of_le (le_of_not_ge h), exact abs_nonneg _, exact is_measurable_interval }
+end
+
+lemma norm_integral_le_abs (a b : ℝ) (f : ℝ → β):
+  ∥(∫ x in a...b, f x)∥ ≤ abs (∫ x in a...b, ∥f x∥) :=
+begin
+  split_ifs with h,
+  { simp only [interval_of_le h] at *,
+    rw abs_of_nonneg,
+    { apply norm_integral_on_le },
+    exact integral_on_nonneg (λ _ _, norm_nonneg _) },
+  { simp only [interval_of_le' (le_of_not_ge h)] at *,
+    rw [abs_of_nonpos, neg_neg, norm_neg],
+    { apply norm_integral_on_le },
+    rw neg_nonpos,
+    exact integral_on_nonneg (λ _ _, norm_nonneg _) }
+end
+
+lemma abs_integral_le_of_ae_le {f g : ℝ → ℝ}
+  (hf : ∀ₘ x, x ∈ a..b → 0 ≤ f x) (hfg : ∀ₘ x, x ∈ a..b → f x ≤ g x)
+  (hgm : measurable_on a..b g) (hgi : integrable_on a..b g) :
+  abs (∫ x in a...b, f x) ≤ abs (∫ x in a...b, g x) :=
+begin
+  by_cases H : measurable_on a..b f ∧ integrable_on a..b f,
```

Comment:
Probably not for this PR: does it make sense to deal with `integrable` but not `measurable` functions? If not, I propose to define `integrable = measurable ∧ old_integrable`.